src/api/app/models/kiwi/image/xml_builder.rb

```diff
@@ -112,6 +113,23 @@ def update_repositories(document)
         document
       end
 
+      def update_profiles(document)
+        comment = document.xpath('comment()[contains(., "OBS-Profiles:")]')
+
+        unless comment.empty?
+          document.xpath('comment()[contains(., "OBS-Profiles:")]').remove
+        end
+
+        # TODO: Scope?
+        selected_profiles = @image.profiles.where(selected: true).pluck(:name)
+        unless selected_profiles.empty?
+          # TODO: The previous comment could have contained a ?variable? (such as @BUILD_FLAVOR@), should it be included again?
```

Comment:
From my POV this is not necessary because:

1) UI shows all profiles of the kiwi xml with the profiles selected you got from the comment
2) You just write the new selection as a comment again (this will contain the previous profiles as well again)

right?

Reply:
The comment is removed before being added again with the new selections. So it removes the variables.

Reply:
If these variables don't bring anything, then I don't mind removing them when rewriting the comment. However, I'm not sure if this is really the case. Could someone confirm?